graviti/openapi/policy.py

```diff
@@ -92,3 +184,58 @@ def get_object_policy(
         params["expired"] = expired
 
     return open_api_do("GET", access_key, url, params=params).json()  # type: ignore[no-any-return]
+
+
+def read_file(  # pylint: disable=inconsistent-return-statements
+    access_key: str,
+    url: str,
+    owner: str,
+    dataset: str,
+    file_name: str,
+    *,
+    allow_retry: bool = True,
+) -> Optional[ByteString]:
+    """Read one remote file.
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        owner: The owner of the dataset.
+        dataset: Name of the dataset, unique for a user.
+        file_name: The name of the file.
+        allow_retry: Whether to allow getting data read permission again.
+
+    Raises:
+        ResponseError: When the status code oss OpenAPI returns is unexpected.
+
+    Returns:
+        The binary content of the file.
+
+    """
+    permission = _get_read_permission(
+        access_key,
+        url,
+        owner,
+        dataset,
+        not allow_retry,
+    )
+    # TODO: add getting data methods for s3 and azure
+    backend_type = permission["backend_type"]
+    policy = permission["policy"]
+    if backend_type == "OSS":
+        try:
+            return _get_binary_file_from_oss(
+                policy["AccessKeyId"],
+                policy["AccessKeySecret"],
+                policy["SecurityToken"],
+                policy["endpoint"],
+                policy["bucket"],
+                file_name,
+            )
+        except ResponseError as error:
+            _error = xmltodict.parse(error.response.text)["Error"]
+            if allow_retry and _error["Code"] == "AccessDenied":
```

Comment:
Could the file be retried after retry with the same params when got `AccessDenied` error?